src/main/java/com/denizenscript/ddiscordbot/commands/DiscordInteractionCommand.java

```diff
@@ -0,0 +1,566 @@
+                            } 
+                            else {
+                                Debug.echoError("Failed to send attachment - missing content?");
+                            }
+                        }
+                        if (action == null) {
+                            Debug.echoError("Failed to send message - missing content?");
+                            scriptEntry.setFinished(true);
+                            return;
+                        }
+                        if (actionRows != null) {
+                            action.addActionRows(actionRows);
+                        }
+                        action.complete();
+                    } else {
+                        ReplyAction action = null;
+                        Interaction replyTo = interaction.getInteraction();
+                        if (embed != null) {
+                            action = replyTo.replyEmbeds(embed);
+                        } 
+                        else {
+                            action = replyTo.reply(message.asString());
+                        }
+                        if (attachFileName != null) {
+                            if (attachFileText != null) {
+                                action = action.addFile(attachFileText.asString().getBytes(StandardCharsets.UTF_8), attachFileName.asString());
+                            } 
+                            else {
+                                Debug.echoError("Failed to send attachment - missing content?");
+                            }
+                        }
+                        if (action == null) {
+                            Debug.echoError("Failed to send message - missing content?");
+                            scriptEntry.setFinished(true);
+                            return;
+                        }
+                        if (actionRows != null) {
+                            action.addActionRows(actionRows);
+                        }
+                        action.setEphemeral(isEphermal);
+                        action.complete();
+                    }
+                    break;
+                }
+                case DELETE: {
+                    if (interaction == null) {
+                        Debug.echoError(scriptEntry.getContext(), "Must specify an interaction!");
+                        scriptEntry.setFinished(true);
+                        return;
+                    } 
+                    else if (interaction.getInteraction() == null) {
+                        Debug.echoError(scriptEntry.getContext(), "Invalid interaction! Has it expired?");
+                        scriptEntry.setFinished(true);
+                        return;
+                    }
+                    interaction.getInteraction().getHook().deleteOriginal().complete();
+                    break;
+                }
+            }
+            scriptEntry.setFinished(true);
```

Comment:
If any exceptions occur, this line will never be reached - so probably add a try/catch

Reply:
Should I just echoError the caught exception?

Reply:
yep